saplings/circuits/src/state/circuits.js

```diff
@@ -297,12 +299,18 @@ function useCircuitState(circuitId) {
         try {
           apiCircuit = await getCircuit(stateCircuitId, user.token);
         } catch (circuitError) {
-          try {
-            apiCircuit = await getProposal(stateCircuitId, user.token);
-          } catch (proposalError) {
-            throw Error(
-              `Unable to fetch ${stateCircuitId} from the splinter daemon`
-            );
+          if (circuitError.code === '401') {
+            window.location.href = `${window.location.origin}/circuits`;
```

Comment:
Not too familiar with this codebase but it seems a bit off for the state handlers to be controlling the view/routing in this way. The page that shows the circuit should probably detect if it isn't able to display a circuit and display an error page instead.

Reply:
Replaced this redirect behavior to just display an error message (with the error from splinterd)